idea/idea-frontend-independent/src/org/jetbrains/kotlin/idea/search/searchUtil.kt

```diff
@@ -166,4 +162,16 @@ fun PsiElement.getKotlinFqName(): FqName? = when (val element = namedUnwrappedEl
     }
     is KtNamedDeclaration -> element.fqName
     else -> null
+}
+
+fun KtDotQualifiedExpression.fqNameSegments(): List<String>? {
```

Comment:
Maybe add a comment with example? I guess it's like: p.q.Outer.Inner.Companion -> [p, q, Outer, Inner, Companion] (or reversed)?